### PR TITLE
[gcp] pkg/asset/machines: fix gcp machine decoding

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

**Comment by csrwng** [comment]
@abhinavdahiya ptal

File: pkg/asset/machines/master.go [hunk 1]
```diff
@@ -264,6 +264,7 @@ func (m *Master) Machines() ([]machineapi.Machine, error) {
 	gcpapi.AddToScheme(scheme)
 	libvirtapi.AddToScheme(scheme)
 	openstackapi.AddToScheme(scheme)
+	gcpprovider.SchemeBuilder.AddToScheme(scheme)
```
Comment (abhinavdahiya) [inline pkg/asset/machines/master.go]: this is already being set [here](https://github.com/openshift/installer/blob/1a78ab5e12aa9295b20c789c0e42c0760bb753c1/pkg/asset/machines/master.go#L264) [api](https://www.godoc.org/github.com/openshift/cluster-api-provider-gcp/pkg/apis#AddToScheme)

File: pkg/asset/machines/gcp/machines.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func provider(clusterID string, platform *gcp.Platform, mpool *gcp.MachinePool,
 	az := mpool.Zones[azIdx]
 	return &gcpprovider.GCPMachineProviderSpec{
 		TypeMeta: metav1.TypeMeta{
-			APIVersion: "gcpproviderconfig.openshift.io/v1beta1",
+			APIVersion: "gcpprovider.openshift.io/v1beta1",
```
Comment (abhinavdahiya) [inline pkg/asset/machines/gcp/machines.go]: can you point to the docs/link which is the source of truth for this..?

Comment (abhinavdahiya) [reply pkg/asset/machines/gcp/machines.go]: hmm maybe https://github.com/openshift/cluster-api-provider-gcp/blob/cb0305f3b986e67ff8d5bb0b375dd7bbb835db01/pkg/apis/gcpprovider/v1beta1/register.go#L6

File: pkg/asset/machines/master.go (hunk 1 above)
Comment (csrwng) [reply pkg/asset/machines/master.go]: ack thx, I think I added this initially when I got the error about unrecognized type, but it's likely that only the group was wrong. Will remove.

File: pkg/asset/machines/gcp/machines.go (hunk 2 above)
Comment (csrwng) [reply pkg/asset/machines/gcp/machines.go]: @abhinavdahiya yes, that's it

**Comment by csrwng** [comment]
@abhinavdahiya comments addressed

**Pull request merged**